app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-copy-tracker.js

```diff
@@ -0,0 +1,55 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analyticsModules || {};
+
+(function (analyticsModules) {
+  'use strict'
+
+  var Ga4CopyTracker = {
+    init: function () {
+      window.addEventListener('copy', this.trackCopy.bind(this))
+    },
+
+    trackCopy: function (event) {
+      try {
+        var text = this.getSelectedText()
+        if (!text) {
+          return // do nothing if no text has been selected
+        }
+        var target = event.target.tagName
+        if (target === 'INPUT' || target === 'TEXTAREA') {
+          return // do nothing if text is being copied from an input
+        }
+
+        var PIIRemover = new window.GOVUK.analyticsGa4.PIIRemover()
+        text = window.GOVUK.analyticsGa4.core.trackFunctions.removeLinesAndExtraSpaces(text)
+        text = PIIRemover.stripPIIWithOverride(text, true, true)
+        var length = text.length
+        text = text.substring(0, 500)
+
+        var data = {
+          event_name: 'copy',
+          type: 'copy',
+          action: 'copy',
+          method: 'browser copy',
+          text: text,
+          copy_length: length
```

Comment:
Non-blocking, is it worth setting `text.length` directly here instead of creating the variable?

Reply:
Excellent question. If we set it here, it's already been trimmed to a maximum of 500 characters (the limit for GA4 attribute data) so we have to store it as a variable earlier. Or store the original text as a variable. Either way, a variable is involved - and we'd like to know the original size of the copied text (which could be thousands of characters, if it's the whole page).

Reply:
Ah, that makes sense - thanks!

Reply:
@andysellick Interesting, none of this crossed my mind! I think what you've done makes sense :+1: